### DISCUSSION TITLE
can this library run on android with .NET MAUI for audio conversion?

### DISCUSSION THREAD

**Discussion by gabrielesilinic** [naudio/NAudio#1051] [Q&A] [answered]
I need to convert a bunch of file formats (mp3 opus and m4a) to a 16 kHz wav file for another library to be able to process it. if the updates that where made on this library made such conversions possible it would just be a dream come true.

**Comment by markheath** [comment, marked as answer]
To access codecs in NAudio you either need to wrap native libraries for your plaftorm (which NAudio does not have for Android), or use fully managed codecs. For MP3, you can look at NLayer, and for opus you can look at Concentus. However, calling into native libraries is always going to be easiest. If there is any .NET FFMPEG integration that works on Android, that would likely be an easier route.

**Reply by gabrielesilinic** [reply]
Yeah the thing is that most ffmpeg wrappers happen to be under a strongly copyleft license and I really don't want my software have such a restrictive license, It will still be open source but if I can get it to stay apache or LGPL at worse it would be nice.

Anyway thank you for the suggestions, I will look into those.
